api/v1alpha1/workspace_validation.go

```diff
@@ -271,7 +271,19 @@ func (r *DataDestination) validateCreate() (errs *apis.FieldError) {
 		re := regexp.MustCompile(`^(.+/[^:/]+):([^:/]+)$`)
 		if !re.MatchString(r.Image) {
 			errs = errs.Also(apis.ErrInvalidValue("Invalid image format, require full output image URL", "Image"))
+		} else {	// Executes if image is of correct format
```

Comment:
Should you do the same check for the data source as well? or maybe have a common function due to the duplicated code.

Reply:
And add UT pls

Reply:
cool added function and added UTs